File: backend/src/main/java/com/sportganise/entities/account/Account.java
package com.sportganise.entities.account;

import com.sportganise.dto.account.AccountPermissions;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/** Entity Model for Account table. */
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Account implements AccountPermissions {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "account_id")
  private Integer accountId;

  @Column(name = "type")
  @Enumerated(EnumType.STRING)
  private AccountType type;

  private String email;

  @Column(name = "auth0_id")
  private String auth0Id;

  private String phone;

  @Column(name = "first_name")
  private String firstName;

  @Column(name = "last_name")
  private String lastName;

  @Column(name = "picture")
  private String pictureUrl;

  @Embedded private Address address;
}

File: frontend/src/types/account.ts
export interface AccountDetailsDirectMessaging {
  accountId: number;
  firstName: string;
  lastName: string;
  pictureUrl: string;
  type: string;
  phone: string;
  selected: boolean | false;
}

export interface Address {
  line: string;
  city: string;
  province: string;
  country: string;
  postalCode: string;
}

export type AccountType = "ADMIN" | "COACH" | "PLAYER" | "GENERAL";

export interface Account {
  accountId: number;
  type: AccountType;
  email: string;
  auth0Id: string;
  address: Address;
  phone: string;
  firstName: string;
  lastName: string;
  pictureUrl: string;
}

export interface AccountPermissions {
  accountId: number;
  email: string;
  firstName: string;
  lastName: string;
  type: string;
  pictureUrl: string;
}

export interface UpdateAccountPayload {
  firstName?: string;
  lastName?: string;
  phone?: string;
  address?: {
    line?: string;
    city?: string;
    province?: string;
    country?: string;
    postalCode?: string;
  };
}
